paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -246,21 +247,27 @@ internal class DefaultFlowController @Inject internal constructor(
     }
 
     override fun confirm() {
-        val state = runCatching {
-            requireNotNull(viewModel.state)
-        }.getOrElse {
-            error(
-                "FlowController must be successfully initialized using " +
-                    "configureWithPaymentIntent() or configureWithSetupIntent() " +
-                    "before calling confirm()"
+        val state = viewModel.state ?: error(
+            "FlowController must be successfully initialized " +
+                "using ${formattedConfigureMethodNames()} before calling confirm()."
+        )
+
+        if (configurationHandler.isConfiguring || viewModel.didLastConfigurationFail) {
+            val error = IllegalStateException(
+                "FlowController.confirm() can only be called if the most recent call " +
+                    "to ${formattedConfigureMethodNames()} has completed successfully."
             )
+            onPaymentResult(PaymentResult.Failed(error))
```

Comment:
This is a behavior change in that we no longer present a potentially outdated sheet, but we fail outright (both of which don’t seem ideal). Should we be more explicit here and introduce custom exceptions that give an indication? For instance, this could be a `PaymentSheetNotYetConfiguredException`.

Reply:
I think this behavior change is fine. New behavior makes sense to me, and shouldn't be a breaking change.